### PR TITLE
Add `esm` ( module ) build output.

### PR DESCRIPTION
**Pull request by gonnavis**
Related discussion: https://discourse.threejs.org/t/three-mesh-bvh-a-plugin-for-fast-geometry-raycasting-and-spatial-queries/26394/2

This will add a `esm` ( module ) bundle file to `./esm/index.js` after run `npm run build`.

In my current project's specific case, I need change 
`import ... from 'three';` to
`import ... from './three/build/three.module.js';` after build.
But I think I can't hard-code the url of three.js.
Is it acceptable to leave `import ... from 'three';` as is and let user to change the url afterwards?

### CODE REVIEW

**Comment by gkjohnson** [comment]
Awesome, thank you!

> In my current project's specific case, I need change `import ... from 'three';` to `import ... from './three/build/three.module.js';` after build. But I think I can't hard-code the url of three.js. Is it acceptable to leave import ... from 'three'; as is and let user to change the url afterwards?

Yes I think that's the best thing to do for now.

File: rollup.config.js [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+	
+		output: {
+	
+			name: 'MeshBVHLib',
+			extend: true,
+			format: 'esm',
+			file: './esm/index.js',
+			sourcemap: true,
+	
+			globals: p => /^three/.test( p ) ? 'THREE' : null,
```
Comment (gkjohnson) [inline rollup.config.js:33]: For the "esm" build I don't think the `name`, `extend`, or `global` options are necessary -- can we remove those?

Comment (gonnavis) [reply rollup.config.js]: Removed and tested that the output is the same!

**Comment by gkjohnson** [comment]
Thanks!

**Pull request merged**